coccinelle/namestr.cocci

```diff
@@ -0,0 +1,25 @@
+// Find direct access of data member of a NameData structure
+//
+// These should use NameStr instead.
+
+@@
+typedef NameData;
+typedef Name;
+NameData E;
+@@
+- E.data
++ NameStr(E)
```

Comment:
In other rules, we also add a comment about the problem to the source code to make the diff output easier to understand (see [this](https://github.com/timescale/timescaledb/blob/main/coccinelle/bms_result.cocci#L9)).

Reply:
Yes, but since these are expressions inside other expressions, it will look really ugly, so I didn't add them, but I am fine with adding them if you think it is useful. I though the change was clear enough anyway.

Reply:
What does it look like with a comment? I think we have a similar one for OidIsValid and it was OK. I'd add it if possible because I'm always confused by these coccinelle checks :)

Reply:
Pushed a commit with a comment just to demonstrate how it looks.